apps/dashboard/src/components/primitives/tabs.tsx

```diff
@@ -22,24 +25,64 @@ const tabsListVariants = cva('inline-flex', {
   },
 });
 
-type TabsListProps = React.ComponentPropsWithoutRef<typeof TabsPrimitive.List> & VariantProps<typeof tabsListVariants>;
+type TabsListProps = React.ComponentPropsWithoutRef<typeof TabsPrimitive.List> &
+  VariantProps<typeof tabsListVariants> & {
+    floatingBgClassName?: string;
+  };
 
 const TabsList = React.forwardRef<React.ElementRef<typeof TabsPrimitive.List>, TabsListProps>(
-  ({ className, variant, align, ...props }, ref) => (
-    <TabsPrimitive.List ref={ref} className={tabsListVariants({ variant, align, className })} {...props} />
-  )
+  ({ className, variant, align, floatingBgClassName, ...props }, forwardedRef) => {
+    const [lineStyle, setLineStyle] = React.useState({ width: 0, left: 0 });
+    const { mounted, listRef } = useTabObserver({
+      onActiveTabChange: (_, activeTab) => {
+        const { offsetWidth: width, offsetLeft: left } = activeTab;
+        setLineStyle({ width, left });
+      },
+    });
+
+    if (variant === 'default' || !variant) {
+      return (
+        <SegmentedControlList
+          ref={forwardedRef}
+          className={tabsListVariants({ variant, align, className })}
+          floatingBgClassName={floatingBgClassName}
+          {...props}
+        />
+      );
+    }
+
+    return (
+      <TabsPrimitive.List
+        ref={mergeRefs(forwardedRef, listRef)}
+        className={tabsListVariants({ variant, align, className })}
+        {...props}
+      >
+        <Slottable>{props.children}</Slottable>
+        <div
+          className={cn('bg-primary absolute bottom-0 left-0 h-[2px] transition-all duration-300', {
+            hidden: !mounted,
+          })}
+          style={{
+            transform: `translate3d(${lineStyle.left}px, 0, 0)`,
+            width: `${lineStyle.width}px`,
+            transitionTimingFunction: 'cubic-bezier(0.65, 0, 0.35, 1)',
+          }}
+          aria-hidden="true"
+        />
+      </TabsPrimitive.List>
```

Comment:
I also updated the regular tabs to have a floating island. 

https://github.com/user-attachments/assets/76c19e08-794f-4bda-9045-45388e9281ed